### PR TITLE
Enrich descriptor from pb with annotations.

### PR DESCRIPTION
**Pull request by wscheep**
Note that there is still a fix needed to import all options in a generic way.

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -28,6 +28,20 @@ protobuf {
     protoc {
         artifact = "com.google.protobuf:protoc:3.6.1"
     }
+    plugins {
```
Comment (alexvanboxel) [inline core/build.gradle]: Please don't add the grpc in the core

File: build.gradle [hunk 2]
```diff
@@ -1,12 +1,19 @@
+/*
```
Comment (alexvanboxel) [inline build.gradle]: Please don't re-add the generated comments that are already removed

File: build.gradle [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+ * user guide available at https://docs.gradle.org/4.10.2/userguide/java_library_plugin.html
+ */
+
 plugins {
     id 'base'
 }
+allprojects {
 
-ext.protobufVersion = "3.6.1"
+    group = "io.anemos.metastore"
```
Comment (alexvanboxel) [inline build.gradle]: Please don't add the group that is defined in the group project, the group is also incorrect

File: core/build.gradle [hunk 4]
```diff
@@ -28,6 +28,20 @@ protobuf {
...
+        grpc {
```
Comment (alexvanboxel) [inline core/build.gradle]: Please don't add the grpc plugin in core

File: core/src/main/java/io/anemos/metastore/core/proto/Convert.java [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 import com.google.protobuf.DescriptorProtos;
 import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.ExtensionRegistry;
+import io.anemos.Annotations;
```
Comment (alexvanboxel) [inline core/src/main/java/io/anemos/metastore/core/proto/Convert.java]: Annotations is deprecated, it's options now

**Comment by wscheep** [comment]
done. Still something fishy with publishing protobeam artifacts with the 'proto' classifier. 

From proto-beam:proto-options
```
task protoJar(type: Jar) {
    classifier = 'proto'
    from('src/main/proto') {
        include '/**/*'
    }
}

when classifier is set, you can compile the dependency as:
```
compile(group: 'io.anemos', name: 'protobeam-options-proto', version: '0.0.3-SNAPSHOT', classifier: 'proto')
```
or 
```
compile('io.anemos:protobeam-options-proto:0.0.3-SNAPSHOT')
```

However, as we need to compile the protos, we want to do protobuf instead of compile:
```
protobuf('io.anemos:protobeam-options-proto:proto:0.0.3-SNAPSHOT')
```
which does not work.

Current workaround: not using proto classifier when building jar.

File: core/build.gradle (hunk 1 above)
Comment (wscheep) [reply core/build.gradle]: done

File: build.gradle (hunk 2 above)
Comment (wscheep) [reply build.gradle]: done

**Pull request merged**